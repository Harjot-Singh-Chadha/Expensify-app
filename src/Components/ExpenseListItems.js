import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
import numeralCurrencyFormat from '../locales/fr';



 const ExpenseListItems = ({description,amount,createdAt,id}) => (

   <div>
   <Link to={`./edit/${id}`}>
       <h2>{description}</h2>
   </Link>
     <p >Amount: {numeral(amount / 100).format('$0,0.00')} - CreatedAt: {moment(createdAt).format('MMMM Do YYYY')}</p>
     
   </div>
  

);

export  default ExpenseListItems;