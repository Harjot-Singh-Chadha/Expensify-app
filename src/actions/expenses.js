import database from '../firebase/firebase';

// ADD_EXPENSE

export const addExpense = (expense) => ({
    type: 'ADD_EXPENSE',
    expense
});

export const startAddExpense = (expenseData = {}) => {
     return (dispatch) => {
        const {
            description = '',
            notes = '',
            amount = 0,
            createdAt = 0 
        } = expenseData;
        const expense = { description, notes, amount, createdAt };
       
      return database.ref('expenses')
        .push(expense)
        .then((ref)=>{
            dispatch(addExpense({
                id:ref.key,
                ...expense
            }))
        .catch((err)=>{
            console.log(err);
        })
        });
     };

};

// REMOVE_EXPENSE

export const removeExpense = ({id} = {}
) => ({
    type: 'REMOVE_EXPENSE',
    id
});

export const startRemoveExpense = ({ id } = {}) =>{
  
    return (dispatch) => {
      
   return database.ref(`expenses/${id}`).remove().then(()=>{
          dispatch(removeExpense({id}))
     });

    }

};

// EDIT_EXPENSE

export const editExpense = (id, updates) => ({
        type: 'EDIT_EXPENSE',
        id,
        updates
    });

 export const startEditExpense = (id, updates) =>{
  
        return (dispatch) => {
        
            return database.ref(`expenses/${id}`).update(updates).then(()=>{
              
              dispatch(editExpense(id, updates))
         });
    
        }
    
    };

//Set_Expenses

export const setExpenses = (expenses) => ({
    type: 'Set_Expenses',
    expenses
});

export const startSetExpense = () => {
     
    return (dispatch) => {
      
     return database.ref('expenses')
    .once('value')
    .then((DataSnapshot)=>{

      const expenses = [] ;
       DataSnapshot.forEach((childSnapshot)=>{
          expenses.push({
            id:childSnapshot.key,
            ...childSnapshot.val()
          })
       })
      dispatch(setExpenses(expenses));
    })
    .catch((err) =>{
         console.log(`Error getting data from the DB`,err);
    })
    
};

};

    