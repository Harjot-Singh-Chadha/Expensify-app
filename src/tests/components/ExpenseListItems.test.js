import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import  ExpenseListItems  from '../../Components/ExpenseListItems';


test("Should render an expense item", () => {
    const wrapper = shallow(<ExpenseListItems {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
   
  });