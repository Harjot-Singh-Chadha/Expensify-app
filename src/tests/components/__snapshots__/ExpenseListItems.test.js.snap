// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render an expense item 1`] = `
<div>
  <Link
    to="./edit/1"
  >
    <h2>
      Gum
    </h2>
  </Link>
  <p>
    Amount: 
    â‚¹1.95
     - CreatedAt: 
    January 1st 1970
  </p>
</div>
`;
